### PR TITLE
Fix #459: support migrating proxies

### LINKED ISSUE #459: No autodiscovery migration (`api-platform-gw:api`)
Try to use Version 1.1.0 and the master branch. Autodiscovery will not migrated. 

`<api-platform-gw:api apiName="${api.autodiscovery.name}" version="${api.autodiscovery.version}" flowRef="api-main" create="false" apikitRef="api-config" doc:name="API Autodiscovery" />`

Blocker !!

### PR DESCRIPTION
**Pull request by afelisatti**
The changes to support proxies are many since a lot of bugs were found along the way while fixing this. Commits have been separated to make reviewing simpler. Here's the breakdown:
- Initial commit adds Gateway migration steps and tasks unit tests, missing from the project, to guarantee this area has coverage and does not break from now on.
- A report entry is added indicating APIs are not identified via an ID and not their name as in Mule 3.
- The main fix is introduced for HTTP proxies along with E2E tests covering their templates (both with and without domains)
- E2E tests are modified to leave migrated apps in the target folder to facilitate reviewing test failures.
- Coverage for RAML/REST proxies is added, the same fix applied, and a name clashing issue resolved to fully support the migration of such proxies.
- Property handling is also fixed to cover the `expression-language` placeholder with Spring attributes, used in Mule 3 proxies.

### CODE REVIEW

**Comment by afelisatti** [comment]
Hi, @holiday-sunrise! The main change here starts after the `Fix #459: apply proxy migration task to apps` commit. The initial commit adds unit tests for the gateway migration steps and tasks that were missing but are required to validate from now on we don't break anything there. I separated everything into commits that can be individually reviewed since the change was big and required lots of testing.

**Comment by holiday-sunrise** [comment]
Ok now it's clear

File: mule-migration-tool-library/src/main/resources/report.yaml [hunk 1]
```diff
@@ -997,6 +997,11 @@ basicStructure:
     type: WARN
     message: "Configuration item in policy YAML does not have property '{propertyName}', it was set by default to value '{defaultValue}'."
 
+apiDiscovery:
+  apiId:
+    type: WARN
+    message: "APIs in Mule 4 have an identifier, the equivalent should be: '{apiName}:{apiVersion}'."
```
Comment (Pgarrett) [inline mule-migration-tool-library/src/main/resources/report.yaml:1003]: Could we add the link to the documentation on Autodiscovery for Mule 4? https://docs.mulesoft.com/api-manager/2.x/configure-autodiscovery-4-task

**Review by Pgarrett — CHANGES_REQUESTED** [review]

File: mule-migration-tool-library/src/main/resources/report.yaml (hunk 1 above)
Comment (holiday-sunrise) [reply mule-migration-tool-library/src/main/resources/report.yaml:1003]: LGTM

**Review by Pgarrett — APPROVED** [review]

**Pull request merged**